src/Forms/GridField/GridField.php

```diff
@@ -1009,9 +1012,14 @@ public function gridFieldAlterAction($data, $form, HTTPRequest $request)
             $state->setValue($fieldData['GridState']);
         }
 
+        // Fetch the store for the "state" of actions (not the GridField)
+        /** @var StateStore $store */
+        $store = Injector::inst()->create(StateStore::class . '.' . $this->getName());
+
         foreach ($data as $dataKey => $dataValue) {
             if (preg_match('/^action_gridFieldAlterAction\?StateID=(.*)/', $dataKey, $matches)) {
-                $stateChange = $request->getSession()->get($matches[1]);
+                $stateChange = $store->load($matches[1]);
+
                 $actionName = $stateChange['actionName'];
```

Comment:
I get a 500 thrown back in my face on this line if I set `AttributeStore` as my default `SilverStripe\Forms\GridField\FormAction\StateStore`. To replicate:
1. Put the following in your YML config
```yml
---
Name: myproject
After: '#gridfieldconfig'
---
SilverStripe\Core\Injector\Injector:
  SilverStripe\Forms\GridField\FormAction\StateStore:
    class: SilverStripe\Forms\GridField\FormAction\AttributeStore
```
2. Access a gridifield in a model admin (I tried the Company and Employee ones from frameworktest)
3. Try Deleting/Archiving an entry via the GridField action.

I get the following error returned to me.
```
[Notice] Undefined index: actionName
POST /admin/test/SilverStripe-FrameworkTest-Model-Employee/EditForm/field/SilverStripe-FrameworkTest-Model-Employee

Line 1023 in /home/maxime/Projects/ss43/vendor/silverstripe/framework/src/Forms/GridField/GridField.php
```

Reply:
Just double checking that you have the admin dependant PR also checked out?

Reply:
Just had a look. I'm running the `pulls/4.3/abstr-action-state` branch of framework and the `pulls/1.3/may-the-state-be-with-you` branch of admin. I made sure I had the latest changed pulled.

Here's the exact request that is returning a 500 for me.

```
POST http://ss43.max.silverstripe.com/admin/test/SilverStripe-FrameworkTest-Model-Company/EditForm/field/SilverStripe-FrameworkTest-Model-Company

Host: ss43.max.silverstripe.com
User-Agent: Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:63.0) Gecko/20100101 Firefox/63.0
Accept: text/html, */*; q=0.01
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate
Referer: http://ss43.max.silverstripe.com/admin/test/
Content-Type: application/x-www-form-urlencoded; charset=UTF-8
X-Pjax: CurrentField
X-Requested-With: XMLHttpRequest
Content-Length: 346
Connection: keep-alive
Cookie: cms-panel-collapsed-cms-menu=false; PHPSESSID=tbpq0ipneqq2m9du0sdak79gke
Pragma: no-cache
Cache-Control: no-cache

action_gridFieldAlterAction%3FStateID%3Dgf_b8fd577d=&SilverStripe-FrameworkTest-Model-Company%5BGridState%5D=%7B%22GridFieldSortableHeader%22%3A%7B%22SortColumn%22%3A%5B%5D%7D%2C%22GridFieldFilterHeader%22%3A%7B%22Columns%22%3Anull%7D%2C%22GridFieldPaginator%22%3A%7B%22currentPage%22%3A1%7D%7D&SecurityID=75e9240c599dcd6b180a4510ee73bf40fa13071b
```

Reply:
Thanks I'll take a look 🙂

Reply:
I updated the admin PR. There was a change that I remember making but I must've `git checkout --`'d the wrong thing

Reply:
Confirmed fixed.